tooling/helium-server/src/server/server.ts

```diff
@@ -30,43 +34,71 @@ export default async function setupHeliumServer(root: string, viteDevServer: any
     app.get('/graphiql', expressPlayground({ endpoint: '/graphql' }));
     // proxying graphql requests in dev environment because of CORS errors
     app.post('/graphql', async (req, res) => {
-      const reqBody = req.body;
+      const { body: reqBody, headers: reqHeaders } = req;
 
-      if (Array.isArray(reqBody)) {
-        reqBody.push({ user: tiInstance.email });
-      } else {
-        reqBody.user = tiInstance.email;
+      // parse request header and pass-thru authToken header
+      const reqAuthToken =
+        reqHeaders[COOKIE_OR_HEADER_NAME_AUTHTOKEN] ||
+        reqHeaders[COOKIE_OR_HEADER_NAME_AUTHTOKEN.toLowerCase()];
```

Comment:
in `@apollo/client` v3.6.9, the custom request header `authToken` set in link context is transformed to lowercase. may not need to handle the lowercase check here in v3.7+

Reply:
quick follow up: 

the latest `staging` brought in `@apollo/client` v3.7.0. Took another look into this after merging from `staging`, still no luck getting the request headers to preserve the original upper case. What I did was to add option `preserveHeaderCase: true` to all instances of `BatchHttpLink`, according to this doc:
https://www.apollographql.com/docs/react/api/link/apollo-link-batch-http#preserveheadercase